ui/src/hooks/useMonitoring.js

```diff
@@ -17,6 +17,9 @@ const getMonitoringLink = (
   routerVersion
 ) => {
   templateSettings.interpolate = /{{([\s\S]+?)}}/g;
+  // If current page is at router details (not viewing a specific router version), the monitoring link will show
+  // metrics for all revisions. Else, the monitoring link will show metrics only for a "specific" revision.
+  routerVersion = /routers\/\d+\/details/.test(window.location.pathname) ? "$__all" : routerVersion
```

Comment:
This looks unnecessarily hack-ish. 

I'd suggest to just update the contract of `getMonitoringLink` and allow `routerVersion` to be `undefined`. And when it's `undefined`, then use `$__all` for a version param.

Then, you can change the calling side and remove `config.version` here:
https://github.com/gojek/turing/blob/b5da2ee2ef5c480e7a6ddbc44b60fd574e31373e/ui/src/router/details/components/page_navigation/RouterDetailsPageNavigation.js#L40

Reply:
Updated